### PR TITLE
[CALCITE-5841] Improve the logic of creating JDBCMeta singleton in calcite plus

### PR DESCRIPTION
**Pull request by chucheng92**
# What is the purpose of the change

Improve the logic of creating JDBCMeta singleton for potential multi-thread problem.

# Brief change log
Add double-checking logic for creating instance.

# Verifying this change

A small work, java behavior ensure its correctness. And also current total cases could cover it.

### CODE REVIEW

File: plus/src/main/java/org/apache/calcite/chinook/ChinookAvaticaServer.java [hunk 1]
```diff
@@ -62,16 +62,18 @@ public static class CalciteChinookMetaFactory implements Meta.Factory {
...
 
     private static JdbcMeta getInstance() {
       if (instance == null) {
-        try {
-          instance =
-              new JdbcMeta(CalciteConnectionProvider.DRIVER_URL,
-                  CONNECTION_PROVIDER.provideConnectionInfo());
-        } catch (SQLException | IOException e) {
-          throw new RuntimeException(e);
+        synchronized (JdbcMeta.class) {
```
Comment (libenchao) [inline plus/src/main/java/org/apache/calcite/chinook/ChinookAvaticaServer.java:69]: How about just synchronizing on `CalciteChinookMetaFactory.class`? (Same to `RawChinookMetaFactory`)

**Review by libenchao — COMMENTED** [review]
@chucheng92 Thanks for your PR, it looks great. I'll left one minor comment.

File: plus/src/main/java/org/apache/calcite/chinook/ChinookAvaticaServer.java (hunk 1 above)
Comment (chucheng92) [reply plus/src/main/java/org/apache/calcite/chinook/ChinookAvaticaServer.java]: @chucheng92 hi, benchao. thanks for reviewing. updated.

**Review by libenchao — APPROVED** [review]
+1, merging

**Comment by libenchao** [comment]
reopened, because we're in the code freeze period for releasing 1.35.0

**Pull request closed without merging**